File: src/main/java/com/kamikazejam/datastore/base/exception/DuplicateCacheException.java
package com.kamikazejam.datastore.base.exception;

import com.kamikazejam.datastore.base.Cache;

@SuppressWarnings("unused")
public class DuplicateCacheException extends CacheException {

    public DuplicateCacheException(Cache<?, ?> cache) {
        super(cache);
    }

    public DuplicateCacheException(String message, Cache<?, ?> cache) {
        super(message, cache);
    }

    public DuplicateCacheException(String message, Throwable cause, Cache<?, ?> cache) {
        super(message, cause, cache);
    }

    public DuplicateCacheException(Throwable cause, Cache<?, ?> cache) {
        super(cause, cache);
    }

    public DuplicateCacheException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace, Cache<?, ?> cache) {
        super(message, cause, enableSuppression, writableStackTrace, cache);
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/base/exception/DuplicateCacheException.kt
package com.kamikazejam.datastore.base.exception

import com.kamikazejam.datastore.base.Cache

@Suppress("unused")
class DuplicateCacheException : CacheException {
    constructor(cache: Cache<*, *>) : super(cache)

    constructor(message: String, cache: Cache<*, *>) : super(message, cache)

    constructor(message: String, cause: Throwable?, cache: Cache<*, *>) : super(message, cause, cache)

    constructor(cause: Throwable?, cache: Cache<*, *>) : super(cause, cache)

    constructor(
        message: String,
        cause: Throwable?,
        enableSuppression: Boolean,
        writableStackTrace: Boolean,
        cache: Cache<*, *>
    ) : super(message, cause, enableSuppression, writableStackTrace, cache)
}
